src/server/db_slice.cc

```diff
@@ -1360,64 +1364,62 @@ pair<uint64_t, size_t> DbSlice::FreeMemWithEvictionStep(DbIndex db_ind, size_t s
   bool record_keys = owner_->journal() != nullptr || expired_keys_events_recording_;
   vector<string> keys_to_journal;
 
-  {
-    FiberAtomicGuard guard;
-    for (int32_t slot_id = num_slots - 1; slot_id >= 0; --slot_id) {
-      for (int32_t bucket_id = num_buckets - 1; bucket_id >= 0; --bucket_id) {
-        // pick a random segment to start with in each eviction,
-        // as segment_id does not imply any recency, and random selection should be fair enough
-        int32_t segment_id = starting_segment_id;
-        for (size_t num_seg_visited = 0; num_seg_visited < max_segment_to_consider;
-             ++num_seg_visited, segment_id = GetNextSegmentForEviction(segment_id, db_ind)) {
-          const auto& bucket = db_table->prime.GetSegment(segment_id)->GetBucket(bucket_id);
-          if (bucket.IsEmpty())
-            continue;
-
-          if (!bucket.IsBusy(slot_id))
-            continue;
-
-          auto evict_it = db_table->prime.GetIterator(segment_id, bucket_id, slot_id);
-          if (evict_it->first.IsSticky() || !evict_it->second.HasAllocated())
-            continue;
-
-          // check if the key is locked by looking up transaction table.
-          const auto& lt = db_table->trans_locks;
-          string_view key = evict_it->first.GetSlice(&tmp);
-          if (lt.Find(LockTag(key)).has_value())
-            continue;
-
-          if (record_keys)
-            keys_to_journal.emplace_back(key);
-
-          evicted_bytes += evict_it->first.MallocUsed() + evict_it->second.MallocUsed();
-          ++evicted_items;
-          PerformDeletion(Iterator(evict_it, StringOrView::FromView(key)), db_table.get());
-
-          // returns when whichever condition is met first
-          if ((evicted_items == max_eviction_per_hb) || (evicted_bytes >= increase_goal_bytes))
-            goto finish;
-        }
-      }
-    }
-  }  // FiberAtomicGuard
+  auto finalize = [&]() {
+    // send the deletion to the replicas.
+    for (string_view key : keys_to_journal) {
+      if (auto journal = owner_->journal(); journal)
+        RecordExpiryBlocking(db_ind, key);
```

Comment:
please add a comment saying that even though you call RecordExpiryBlocking it does not block due to JournalFlushGuard above

Reply:
I find funny that Adi asked me why I used 

```
  // Disable flush journal changes to prevent preemtion
  journal::JournalFlushGuard journal_flush_guard(shard_owner()->journal());
```

when the `caller` of this function already does the same. Then you are asking me to add a comment here.

yet the only reason I added the line above on the `first line of this function` is precisely to `show intent` (that we disable the flushing in journal explicitly). 

I will add a comment :smile: